File: web/src/api/admin/gost_client_host.js
import request from "../../request/index.js";

const baseUrl = '/v1/admin/gost/client/host'

export const apiAdminGostClientHostList = () => {
    return request.request({
        url: `${baseUrl}/list`,
        method: 'POST',
    })
}

export const apiAdminGostClientHostPage = (data) => {
    return request.request({
        url: `${baseUrl}/page`,
        method: 'POST',
        data
    })
}

export const apiAdminGostClientHostCreate = (data) => {
    return request.request({
        url: `${baseUrl}/create`,
        method: 'POST',
        data
    })
}

export const apiAdminGostClientHostConfig = (data) => {
    return request.request({
        url: `${baseUrl}/config`,
        method: 'POST',
        data
    })
}

export const apiAdminGostClientHostUpdate = (data) => {
    return request.request({
        url: `${baseUrl}/update`,
        method: 'POST',
        data
    })
}


export const apiAdminGostClientHostDelete = (data) => {
    return request.request({
        url: `${baseUrl}/delete`,
        method: 'POST',
        data
    })
}

File: server/router/admin/gost_client_host.go
package admin

import (
	"github.com/gin-gonic/gin"
	"server/controller/admin/gost_client_host"
	"server/global"
	"server/router/middleware"
)

func InitGostClientHost(group *gin.RouterGroup) {
	g := group.Group("gost/client/host", middleware.Auth(global.Jwt), middleware.AuthAdmin())
	g.POST("create", gost_client_host.Create)
	g.POST("config", gost_client_host.Config)
	g.POST("delete", gost_client_host.Delete)
	g.POST("page", gost_client_host.Page)
}
